test/iidm/CMakeLists.txt

```diff
@@ -58,7 +58,7 @@ set(UNIT_TEST_SOURCES
 
 add_executable(unit-tests-iidm ${UNIT_TEST_SOURCES})
 target_compile_definitions(unit-tests-iidm PRIVATE BOOST_TEST_DYN_LINK)
-target_link_libraries(unit-tests-iidm PRIVATE iidm-tests)
+target_link_libraries(unit-tests-iidm PRIVATE iidm-tests Threads::Threads)
```

Comment:
I think this should be add to the IIDM library instead and then inherited by the other targets?